### ISSUE TITLE
RUF075 triggers on Fastapi documented lifespan code

### ISSUE THREAD

**Issue by m00rex** [astral-sh/ruff#25428]
### Question

I just used `ruff check --preview` on our codebase.
It now triggers on this Fastapi lifespan function:

```python
@asynccontextmanager
async def lifespan(app: FastAPI):
    await _startup(app=app)
    yield
    await _shutdown()
```

```bash
error[RUF075]: Context manager does not catch exceptions
  --> app/api/main.py:43:5
   |
41 | async def lifespan(app: FastAPI):
42 |     await _startup(app=app)
43 |     yield
   |     ^^^^^
44 |     await _shutdown()
   |
```
I understand why it triggers but the official fastapi docs does not use `try-finally` https://fastapi.tiangolo.com/advanced/events/
So i want to ask if the fastapi docs is inaccurate or if it is a `RUF075` false positive?

### Version

0.15.14

**Comment by ntBre** [comment]
I'm not familiar with the internals of FastAPI, but I would still expect the reasoning for the rule to apply in this situation. It may be worth asking on the FastAPI side too!

**Comment by m00rex** [comment]
I opened a discussion over at fastapi: https://github.com/fastapi/fastapi/discussions/15633
